Source/MQTTnet/Formatter/MqttPacketFormatterAdapter.cs

```diff
@@ -120,6 +120,14 @@ MqttProtocolVersion ParseProtocolVersion(ReceivedMqttPacket receivedMqttPacket)
         {
             if (receivedMqttPacket == null) throw new ArgumentNullException(nameof(receivedMqttPacket));
 
+            if (receivedMqttPacket.Body.Length < 7)
+            {
+                // 2 byte protocol name length
+                // at least 4 byte protocol name
+                // 1 byte protocol level
+                throw new MqttProtocolViolationException("Mqtt Connect packet must have at least 7 bytes");
```

Comment:
Please consider writing packets in upper case all the time. I usually do this in all trace messages so that it looks like "MQTT CONNECT packet must ...". The idea behind this is to have a more natural reading flow etc.